File: app/src/main/java/dev/ragnarok/fenrir/util/ValidationUtil.java
package dev.ragnarok.fenrir.util;

import static dev.ragnarok.fenrir.util.Utils.trimmedIsEmpty;

import android.annotation.SuppressLint;

import androidx.core.util.PatternsCompat;

public class ValidationUtil {

    @SuppressLint("RestrictedApi")
    public static boolean isValidURL(String url) {
        return url != null && PatternsCompat.AUTOLINK_WEB_URL.matcher(url).find();
    }

    public static boolean isValidIpAddress(String ipv4) {
        if (trimmedIsEmpty(ipv4)) {
            return false;
        }

        ipv4 = ipv4.trim();

        String[] blocks = ipv4.split("\\.");

        if (blocks.length != 4) {
            return false;
        }

        for (String block : blocks) {
            try {
                int num = Integer.parseInt(block);

                if (num > 255 || num < 0) {
                    return false;
                }
            } catch (Exception e) {
                return false;
            }
        }

        return true;
    }
}

File: app/src/main/java/dev/ragnarok/fenrir/util/ValidationUtil.kt
package dev.ragnarok.fenrir.util

import android.annotation.SuppressLint
import androidx.core.util.PatternsCompat
import dev.ragnarok.fenrir.trimmedIsNullOrEmpty

object ValidationUtil {

    @SuppressLint("RestrictedApi")
    fun isValidURL(url: String?): Boolean {
        return url != null && PatternsCompat.AUTOLINK_WEB_URL.matcher(url).find()
    }


    fun isValidIpAddress(ipv4: String?): Boolean {
        ipv4 ?: return false
        var ipv4T = ipv4
        if (ipv4.trimmedIsNullOrEmpty()) {
            return false
        }
        ipv4T = ipv4T.trim { it <= ' ' }
        val blocks = ipv4T.split("\\.").toTypedArray()
        if (blocks.size != 4) {
            return false
        }
        for (block in blocks) {
            try {
                val num = block.toInt()
                if (num > 255 || num < 0) {
                    return false
                }
            } catch (e: Exception) {
                return false
            }
        }
        return true
    }
}